plugins/chat/assets/javascripts/discourse/components/chat/drawer-routes/browse.gjs

```diff
@@ -0,0 +1,63 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { array } from "@ember/helper";
+import { service } from "@ember/service";
+import i18n from "discourse-common/helpers/i18n";
+import I18n from "I18n";
+import BrowseChannels from "discourse/plugins/chat/discourse/components/browse-channels";
+import Navbar from "discourse/plugins/chat/discourse/components/chat/navbar";
+
+export default class ChatDrawerRoutesBrowse extends Component {
+  @service chat;
+  @service chatStateManager;
+  @service chatChannelsManager;
+  @service chatHistory;
+
+  @tracked showThreadFullTitle = false;
+
+  get showfullTitle() {
+    return this.chatStateManager.isDrawerExpanded && this.showThreadFullTitle;
+  }
+
+  get backButton() {
+    const link = {
+      models: this.chat.activeChannel?.routeModels,
+    };
+
+    link.title = I18n.t("chat.return_to_channel");
+    link.route = "chat.channel";
+
+    return link;
+  }
+
+  <template>
+    <div class="c-drawer-routes --browse">
+      <Navbar
+        @onClick={{this.chat.toggleDrawer}}
+        @showFullTitle={{this.showfullTitle}}
+        as |navbar|
+      >
+        <navbar.BackButton
+          @title={{this.backButton.title}}
+          @route={{this.backButton.route}}
+          @routeModels={{this.backButton.models}}
```

Comment:
As above, can probably remove as I don't think `routeModels` is necessary for the browse route.